### PR TITLE
Document symbol improvements

### LINKED ISSUE #1172: When a server returns SymbolInformation[] in textDocument/documentSymbol, show the containerName
Document symbols can and should support the containerName in the UI.

https://microsoft.github.io/language-server-protocol/specifications/specification-3-15/#textDocument_documentSymbol

`/**
	The name of the symbol containing this symbol. This information is for
	 user interface purposes (e.g. to render a qualifier in the user interface
	 if necessary). It can't be used to re-infer a hierarchy for the document
	 symbols.
	 */
	containerName?: string;`

For example something like this would be helpful to give context to the symbol without affecting the filtering behavior. 

Screen_Shot_2020-06-30_at_12 16 15_PM

### PR DESCRIPTION
**Pull request by rwols**
- Put the name (without spaces) in the first row
- Put the parent hierarchy in the second row
- Put the kind and detail in the third row

~~This only works when the server returns DocumentSymbol[] as response,
not when it returns SymbolInformation[].~~ Works for both DocumentSymbol[] and SymbolInformation[] return types.

close #1172

### CODE REVIEW

File: plugin/symbols.py [hunk 1]
```diff
@@ -119,28 +126,27 @@ def process_symbols(self, items: List[Dict[str, Any]]) -> List[List[str]]:
...
-        children = item.get('children') or []
-        for child in children:
-            self.process_document_symbol_recursive(quick_panel_items, child, depth + 1)
+        name = item['name']
+        with _additional_name(names, name):
+            kind_and_detail = format_symbol_kind(kind)
+            detail = item.get('detail')
+            if detail:
+                kind_and_detail += " - {}".format(detail)
+            quick_panel_items.append([name, " → ".join(names), kind_and_detail])
```
Comment (rchl) [inline plugin/symbols.py:146]: There is some redundancy for top-level symbols but no big deal.

Screenshot 2020-07-18 at 15 46 26

**Comment by rwols** [comment]
This last commit compacts the items to two rows (instead of three rows).

clangd: (DocumentSymbol)
Screenshot from 2020-07-19 21-32-12

pyls: (SymbolInformation)
Screenshot from 2020-07-19 21-31-51

**Comment by AmjadHD** [comment]
doesn't the quick panel support icons ?

**Comment by AmjadHD** [comment]
@rwols doesn't the quick panel support icons?

**Comment by rwols** [comment]
> @rwols doesn't the quick panel support icons?

Unfortunately, no.

**Comment by AmjadHD** [comment]
well that's no right ! you should open a feature request.

**Comment by rwols** [comment]
@AmjadHD great minds think alike https://github.com/sublimehq/sublime_text/issues/3489

**Pull request merged**